blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunContainerImpl.java

```diff
@@ -38,30 +39,12 @@ public Link getLink() {
 
     @Override
     public BlueRun get(String name) {
-        RunList<? extends hudson.model.Run> runList = job.getBuilds();
-
-        hudson.model.Run run = null;
-        if (name != null) {
-            for (hudson.model.Run r : runList) {
-                if (r.getId().equals(name)) {
-                    run = r;
-                    break;
-                }
-            }
-            if (run == null) {
-                throw new ServiceException.NotFoundException(
-                    String.format("Run %s not found in organization %s and pipeline %s",
-                        name, pipeline.getOrganization(), job.getName()));
-            }
-        } else {
-            run = runList.getLastBuild();
-        }
-        return  AbstractRunImpl.getBlueRun(run, pipeline);
+        return RunLoader.get().getRun(name, job, pipeline);
     }
 
     @Override
     public Iterator<BlueRun> iterator() {
-        return RunSearch.findRuns(job, pipeline.getLink()).iterator();
+        return RunLoader.get().getRuns(job, pipeline.getLink()).iterator();
```

Comment:
@vivek  One thing this broke is the executing runs now don't show on /activity. Since they are continuously updating we can't put them in the index.